examples/complete/main.tf

```diff
@@ -270,6 +271,9 @@ module "dedicated_host" {
 #############################################################################
 
 module "slz_vsi_dh" {
+  count                           = var.enable_dedicated_host ? 1 : 0
+  depends_on                      = [module.dedicated_host]
+  dedicated_host_id               = try(module.dedicated_host.dedicated_host_ids[0], null)
```

Comment:
Does this work? I would have went with:
`dedicated_host_id = var.enable_dedicated_host ? module.dedicated_host.dedicated_host_ids[0] : null`

Reply:
I have made the change as you suggested but imho if we are providing `count = var.enable_dedicated_host ? 1 : 0` then only `dedicated_host_id` would come into picture. Thinking this I had not provided the explicit conditional check and `try()` was not needed I agree, added a thought that the default `null` will be assigned if by any chance no value is received.

Reply:
`try()` means it will always try once and might fail silently and default to default value. Where as ternary will always pass on first try. One less transaction

Reply:
I understood the point here. Marking the conversation as resolved.